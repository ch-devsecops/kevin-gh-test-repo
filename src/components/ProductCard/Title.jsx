import React from 'react';
import PropTypes from 'prop-types';
import { MarkdownHeading } from '@honda-canada/design-system-react';

const Title = ({ title, mt }) => (
  <MarkdownHeading mb={['xs', 's']} mt={mt ?? ['s', 'default']} headingOverride="h5">
    {title}
  </MarkdownHeading>
);

Title.propTypes = {
  title: PropTypes.string,
};

export default Title;
