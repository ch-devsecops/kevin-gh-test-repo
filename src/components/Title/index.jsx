import React from 'react';
import { Box, H3, Markdown, useMediaQueries } from '@honda-canada/design-system-react';
import { withSitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import CTA from '../CTA';
import {
  colourTokenForParam,
  getCtaTypeFromBgColor,
  mapGTMCategory,
  foregroundColourTokenForParam,
} from '../../utils/sitecoreFields';
import { wrapJSSFields } from '../../utils/wrapJSSFields';
import { getTitleComponent, stripMarkdownHeading, styledCompiler } from '../../utils/markdown';
import { getComponentFromRoute } from '../../utils/sitecorePlaceholders';
import { getGtmTagValue } from '../../utils/gtmEvents';

const getCTA = (bgColor, ctaUrl, ctaType, ctaIcon, gtmInfo) => {
  const ctaTypeFieldValue = ctaType?.field?.value;
  const type = getCtaTypeFromBgColor(bgColor, ctaTypeFieldValue);
  const ariaLabel = ctaUrl.hasProp('title') ? ctaUrl.getProp('title') : ctaUrl.getProp('text');
  return (
    <CTA
      linkField={ctaUrl}
      typeField={{ value: type }}
      iconField={ctaIcon?.field}
      iconColor={type === 'tertiaryWhite' ? 'white' : undefined}
      gtmTags={{
        'aria-label': ariaLabel,
        'data-gtm-model': getGtmTagValue(gtmInfo?.gtmModelName),
        'data-gtm-trim': getGtmTagValue(gtmInfo?.gtmTrimName),
        'data-gtm-body-style': getGtmTagValue(gtmInfo?.gtmBodyStyle),
        'data-gtm-interaction-type': getGtmTagValue(gtmInfo?.gtmInteractionType),
        'data-gtm-title': getGtmTagValue(gtmInfo?.gtmTitle),
      }}
    >
      {ctaUrl.getProp('description')}
    </CTA>
  );
};

const Title = ({ fields, params, rendering, sitecoreContext }) => {
  const { isMobile } = useMediaQueries();

  if (!fields) {
    return null;
  }

  const {
    seoH1,
    ctaType1,
    ctaUrl1,
    ctaIcon1,
    ctaType2,
    ctaUrl2,
    ctaIcon2,
    ctaGtmInteractionType1,
    ctaGtmModelName1,
    ctaGtmTrimName1,
    ctaGtmBodyStyle1,
    ctaGtmInteractionType2,
    ctaGtmModelName2,
    ctaGtmTrimName2,
    ctaGtmBodyStyle2,
    title,
    description,
    anchorId,
    gtmTitle,
    gtmCategory,
    hideOnMobile,
    bottomPadding,
    topPadding,
  } = wrapJSSFields(fields);

  // If no Markdown heading is provided, the title should be an H3
  const TitleComponent = getTitleComponent(title?.value, H3);

  // FIXME: This ugly little hack makes the title sentence-case instead of all caps
  // if there's a TrimSpecifications component on the page. See https://jira.web.honda.ca:8443/browse/SCC-1561.
  // The proper solution is to not use a Title component on that page, and make the
  // title part of the TrimSpecifications component.
  const tabsContainerComponent = getComponentFromRoute('TabsContainer', sitecoreContext.route);
  const firstTabComponent = tabsContainerComponent?.placeholders['tabs-content'].find(
    component => component.componentName === 'Tab',
  );
  const tabContent = firstTabComponent && firstTabComponent.placeholders['tab-content'];
  const hasTrimSpecificiationsComponent = Boolean(
    tabContent && tabContent.find(c => c.componentName === 'TrimSpecifications'),
  );

  const titleContent = title?.value && (
    <TitleComponent
      color={foregroundColourTokenForParam[params?.bgColour?.toLowerCase()]}
      as={seoH1?.value === 'Title' && 'h1'}
      mb={['xs', 'm']}
      style={{ textTransform: hasTrimSpecificiationsComponent && 'none' }}
    >
      {styledCompiler(stripMarkdownHeading(title?.value))}
    </TitleComponent>
  );

  const bgColour = params?.bgColour?.toLowerCase();
  const CTA1 =
    ctaUrl1.hasProp('href') &&
    ctaUrl1.hasProp('text') &&
    getCTA(bgColour, ctaUrl1, ctaType1, ctaIcon1, {
      gtmTitle: gtmTitle?.value,
      gtmInteractionType: ctaGtmInteractionType1?.value,
      gtmModelName: ctaGtmModelName1?.value,
      gtmTrimName: ctaGtmTrimName1?.value,
      gtmBodyStyle: ctaGtmBodyStyle1?.value,
    });
  const CTA2 =
    ctaUrl2.hasProp('href') &&
    ctaUrl2.hasProp('text') &&
    getCTA(bgColour, ctaUrl2, ctaType2, ctaIcon2, {
      gtmTitle: gtmTitle?.value,
      gtmInteractionType: ctaGtmInteractionType2?.value,
      gtmModelName: ctaGtmModelName2?.value,
      gtmTrimName: ctaGtmTrimName2?.value,
      gtmBodyStyle: ctaGtmBodyStyle2?.value,
    });

  const gtmTags = {
    type: rendering?.componentName,
    category: mapGTMCategory(gtmCategory),
  };
  // eslint-disable-next-line react/no-unstable-nested-components
  const Wrapper = ({ children }) =>
    hideOnMobile?.value && isMobile ? <div style={{ display: 'none' }}>{children}</div> : <div>{children}</div>;

  return (
    <Wrapper>
      <Box
        backgroundColor={colourTokenForParam[params?.bgColour?.toLowerCase()]}
        display="flex"
        justifyContent="center"
        textAlign="center"
        pb={bottomPadding?.value ? `${bottomPadding.value}px` : ['xl', 'big']}
        pt={topPadding?.value ? `${topPadding.value}px` : ['xl', 'big']}
        data-gtm-category={getGtmTagValue(gtmTags?.category)}
        data-gtm-component-type={getGtmTagValue(gtmTags.type)}
        id={anchorId.value}
      >
        <Wrapper>
          <Box>
            {titleContent}
            <Box maxWidth="824px" mx="auto">
              <Markdown
                size="regular"
                color={foregroundColourTokenForParam[params?.bgColour?.toLowerCase()]}
                mb={['xs', 'm']}
                mx={['xs', 's', 'zero']}
                options={
                  (seoH1?.value === 'Description' ? 'h1' : '') && {
                    overrides: {
                      p: {
                        component: ({ children, type, ...props }) => {
                          const Elem = type;
                          return <Elem {...props}>{children}</Elem>;
                        },
                        children: description.value,
                        props: {
                          className: 'override',
                          type: seoH1?.value === 'Description' ? 'h1' : '',
                        },
                      },
                    },
                  }
                }
              >
                {description?.value}
              </Markdown>
            </Box>
            <Box display="flex" justifyContent="center" flexDirection={['column', 'row']}>
              {CTA1 && (
                <Box mt={['xs', 'm']} mx={['zero', 's']}>
                  {CTA1}
                </Box>
              )}
              {CTA2 && (
                <Box mt={['xs', 'm']} mx={['zero', 's']}>
                  {CTA2}
                </Box>
              )}
            </Box>
          </Box>
        </Wrapper>
      </Box>
    </Wrapper>
  );
};

export default withSitecoreContext()(Title);
