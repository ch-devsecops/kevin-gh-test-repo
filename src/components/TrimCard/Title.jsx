import React, { useRef } from 'react';
import { H5, Box, Image } from '@honda-canada/design-system-react';
import { compiler } from 'markdown-to-jsx';
import Link from './Link';
import themeStyles from './TrimCard.styles';

const StyledH5 = themeStyles.apply(H5, 'StyledH5');
const TrimCardImageLink = themeStyles.apply(Link, 'TrimCardImageLink');

const TrimName = ({ name, modelYear, showModelYear }) => (
  <>
    {showModelYear && <Box data-testid="cy-trim-model-year">{compiler(modelYear)}</Box>}
    <Box data-testid="cy-trim-specification-card-name">{compiler(name)}</Box>
  </>
);

const TitleComponent = ({ detailsPath, name, modelYear, nameBadge, isDark, gtmTags, showModelYear }) => {
  const styledH5Ref = useRef();

  return nameBadge?.src ? (
    <Box height="24px" py="xxs">
      {detailsPath ? (
        <TrimCardImageLink {...gtmTags} to={detailsPath}>
          <Image {...nameBadge} />
        </TrimCardImageLink>
      ) : (
        <Image {...nameBadge} />
      )}
    </Box>
  ) : (
    <StyledH5 ref={styledH5Ref} color={isDark ? 'white' : undefined}>
      {detailsPath ? (
        <TrimCardImageLink {...gtmTags} to={detailsPath}>
          <TrimName name={name} modelYear={modelYear} showModelYear={showModelYear} />
        </TrimCardImageLink>
      ) : (
        <TrimName name={name} modelYear={modelYear} showModelYear={showModelYear} />
      )}
    </StyledH5>
  );
};

export default TitleComponent;
