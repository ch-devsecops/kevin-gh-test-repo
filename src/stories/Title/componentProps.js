import mockData from '../../../sample-mock-data/routes/title/en.json';

export default mockData;
