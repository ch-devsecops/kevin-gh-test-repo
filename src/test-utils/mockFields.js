/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
