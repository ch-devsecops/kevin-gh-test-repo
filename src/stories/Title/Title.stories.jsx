import React from 'react';
import Title from '../../components/Title';
import props from './componentProps';

export default {
  title: 'Title',
  component: Title,
};

const Template = args => <Title {...args} />;

export const Default = Template.bind({});
Default.args = props.placeholders['jss-main'][0];
