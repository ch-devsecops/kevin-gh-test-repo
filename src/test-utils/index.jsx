import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
