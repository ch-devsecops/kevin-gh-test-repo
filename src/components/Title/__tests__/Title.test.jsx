import React from 'react';
import { jssRender } from 'test-utils';
import Title from '..';

const routeData = require('../../../../sample-mock-data/routes/title/en.json');

describe('Title snapshots', () => {
  it('renders correct colours with blue background', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][0]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders correct colours with black background', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][1]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders correct colours with red background', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][2]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders correct colours with white background', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][3]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders correct colours with honda black background', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][4]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders with one CTA', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][5]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders with the specified bottomPadding and topPadding', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][6]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders with tertiary CTAs with cta icons', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][7]} />);

    expect(container).toMatchSnapshot();
  });

  it('renders white tertiary CTAs when the background color is dark', () => {
    const { container } = jssRender(<Title {...routeData.placeholders['jss-main'][8]} />);

    expect(container).toMatchSnapshot();
  });
});
