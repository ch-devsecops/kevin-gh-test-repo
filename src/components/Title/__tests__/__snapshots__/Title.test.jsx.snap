// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title snapshots renders correct colours with black background 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 dsiqAZ"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 kUJimD"
            color="white"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 ciyLLm"
              color="white"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fBGdmK"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fDIHTQ"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://example.com#linktwo"
                title=""
              >
                Title Link Two
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders correct colours with blue background 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 ibAYVY"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 kUJimD"
            color="white"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title Lorem ipsum dolor sit amet, consectetur adipiscing elit
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 ciyLLm"
              color="white"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fDIHTQ"
                data-gtm-body-style="some body style"
                data-gtm-interaction-type="cta: link1"
                data-gtm-model="some model name"
                data-gtm-title="gtm-title"
                data-gtm-trim="some trim name"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One...
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fDIHTQ"
                data-gtm-interaction-type="cta: link2"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://example.com#linktwo"
                title=""
              >
                Title Link Two...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders correct colours with honda black background 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 eRqaNZ"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 kUJimD"
            color="white"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 ciyLLm"
              color="white"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                This is my new component
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fBGdmK"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fDIHTQ"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://example.com#linktwo"
                title=""
              >
                Title Link Two
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders correct colours with red background 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 gEjPLB"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 kUJimD"
            color="white"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 ciyLLm"
              color="white"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fBGdmK"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fDIHTQ"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://example.com#linktwo"
                title=""
              >
                Title Link Two
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders correct colours with white background 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 dHDoXS"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 dVbPXE"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 fBZbFP"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fePOwb"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 gvXiRa"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://example.com#linktwo"
                title=""
              >
                Title Link Two
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders white tertiary CTAs when the background color is dark 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 dsiqAZ"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 kUJimD"
            color="white"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 ciyLLm"
              color="white"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <span>
                <a
                  aria-label=""
                  class="Link-sc-16wd4a0-0 dKSZAU"
                  data-gtm-body-style="some body style"
                  data-gtm-interaction-type="cta: link1"
                  data-gtm-model="some model name"
                  data-gtm-title="gtm-title"
                  data-gtm-trim="some trim name"
                  href="/specs"
                  title=""
                >
                  Title Link One
                </a>
              </span>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <span>
                <a
                  aria-label=""
                  class="Link-sc-16wd4a0-0 dKSZAU"
                  data-gtm-interaction-type="cta: link2"
                  data-gtm-title="gtm-title"
                  href="https://example.com#linktwo"
                  title=""
                >
                  Title Link Two
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders with one CTA 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 eOWWpR"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 kUJimD"
            color="white"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 ciyLLm"
              color="white"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                This is my new component
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fDIHTQ"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders with tertiary CTAs with cta icons 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 gcxozH"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 cZlNZm"
            color="black"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 fGBMGh"
              color="black"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <span>
                <a
                  aria-label=""
                  class="Link-sc-16wd4a0-0 eYsnbh"
                  data-gtm-body-style="some body style"
                  data-gtm-interaction-type="cta: link1"
                  data-gtm-model="some model name"
                  data-gtm-title="gtm-title"
                  data-gtm-trim="some trim name"
                  href="/specs"
                  title=""
                >
                  Title Link One
                </a>
              </span>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <span>
                <a
                  aria-label=""
                  class="Link-sc-16wd4a0-0 eYsnbh"
                  data-gtm-interaction-type="cta: link2"
                  data-gtm-title="gtm-title"
                  href="https://example.com#linktwo"
                  title=""
                >
                  Title Link Two
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title snapshots renders with the specified bottomPadding and topPadding 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 bYRSxu"
      data-gtm-category="Others"
      display="flex"
      id="TitleAnchorId"
    >
      <div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <h3
            class="sc-1k44dfv-0 cZlNZm"
            color="black"
          >
            <span>
               Example
              <sup>
                tm
              </sup>
               Title
            </span>
          </h3>
          <div
            class="Box-sc-1bsf784-0 cFYsYd"
          >
            <div
              class="sc-1k44dfv-0 fGBMGh"
              color="black"
              mb="xs,m"
              mx="xs,s,zero"
            >
              

              <h1
                class="override"
              >
                Example Description test test test 1-888-556-8548
              </h1>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 epXBQL"
            display="flex"
          >
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 fePOwb"
                data-gtm-body-style="some body style"
                data-gtm-interaction-type="cta: link1"
                data-gtm-model="some model name"
                data-gtm-title="gtm-title"
                data-gtm-trim="some trim name"
                data-testid="dsr-button-component"
                href="https://google.com#linkone"
                title=""
              >
                Title Link One...
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 erzPKL"
            >
              <a
                aria-label=""
                class="Button-pulw6t-0 gIftGV"
                data-gtm-interaction-type="cta: link2"
                data-gtm-title="gtm-title"
                data-testid="dsr-button-component"
                href="https://example.com#linktwo"
                title=""
              >
                Title Link Two...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
